### DISCUSSION TITLE
Support for Enhanced RTMP

### DISCUSSION THREAD

**Discussion by mat-hek** [membraneframework/membrane-core#882] [Ideas]
Originally posted by @andr-ec in https://github.com/membraneframework/membrane_core/issues/881

> This probably implies support for AV1 and HEVC.

> It looks like someone already has an implementation of HEVC for membrane: https://github.com/gBillal/membrane_h265_ffmpeg_plugin

**Comment by mat-hek** [comment]
It's not in the roadmap yet, but when (if?) the adoption of Enhanced RTMP grows, we'll definitely give it a close look. H265 is indeed supported thanks to the someone, namely Billal ;) AV1 is not supported yet but it's on our roadmap.

For now, personally, this is how enhanced RTMP looks like to me 😄 

![image](https://github.com/user-attachments/assets/c88a14c6-d9ec-4189-a889-f0807e9664e6)

Simplicity is the best part of RTMP, and now instead of using something actually better (like SRT) they add more stuff to RTMP which doesn't quite fit ¯\\_(ツ)\_/¯ But I suppose there are some good reasons for it, let's see how the adoption goes.

**Comment by andr-ec** [comment]
Thanks for the response @mat-hek !!

I definitely get that. 

As far as I understand, YouTube supports it and ant media supports it. Twitch is working on support.

What level of adoption are we looking for? just for my information.

And are there any ways to handle these next gen formats (HEVC and av1) in a stream currently with membrane?

**Comment by mat-hek** [comment]
Our main metric of adoption is actually people asking and (lack of) _X_ being a blocker in people's use cases :) Twitch support would be an important factor too.

Regarding H265, you can stream via RTSP and RTP, possibly also WebRTC, but we haven't tried it. Also, there's encoder and decoder, and MP4 supports H265. AV1 is not supported at all - that's hopefully going to change soon.

**Comment by andr-ec** [comment]
That's helpful thanks. And if I was to tackle implementing enhanced rtmp on my own, do you have any suggested resources?

**Comment by mat-hek** [comment]
Hmm, so there's obviously the specification: https://veovera.org/docs/enhanced/enhanced-rtmp-v1 and it looks quite straightforward. There's also [v2 version](https://veovera.org/docs/enhanced/enhanced-rtmp-v2), but it's marked as 'alpha', so maybe it's worth going for v1. Apart from more video codecs, the specification allows for more audio codecs and multi-track support, which don't have to be there to have something useful IMO. Apart from the specs, I can't recommend any materials :/
